test/integration/test/no_travis/pagekite_ping_test.js

```diff
@@ -7,59 +7,69 @@ var config = new Config('./test/integration/lib/config/foxbox.js');
 var Prepper = require('../../lib/testPrepper.js');
 
 Prepper.makeSuite('Verify validity of pagekite ping endpoint', function(){
-  var local_ip,tunnel_url;
+  var local_url,tunnel_url;
   var header = new Config('./test/integration/lib/config/header.js');
   var credential = config.get('credential'); 
-
-  it('get address from pagekite ping endpoint', function() {
-    var pingUrl = config.get('pagekite.r') + '/ping';
+  var pingUrl = config.get('pagekite.r') + '/ping';
+  it('get address from pagekite ping endpoint', function() {  
     var pick;
     var timestamp = 0;
 
     return  chakram.get(pingUrl,{'json': true})
     .then(function(pingResp){
       for (var match in pingResp.body) {
-        if (pingResp.body[match].tunnel_url === 
-          config.get('pagekite.externalURL')) {
+        // may be multiple entries.  in that case, pick latest
+        if (parseInt(pingResp.body[match].timestamp) > 
+          parseInt(timestamp)) {
+          timestamp = pingResp.body[match].timestamp;
+          pick = match;
+        }
+    }
+    
+    expect(pingResp).to.have.status(200);
+    expect(pingResp.body[pick].timestamp).to.match(/\d+/);
+    expect(pingResp.body[pick].public_ip).to.match(/\d+.\d+.\d+.\d+/);
+    });
+  });
+  
+  describe('Initiate the remote connection',function(){
+    before('fetch the url', function() {
+      var pick;
+      var timestamp = 0;
+
+      return  chakram.get(pingUrl,{'json': true})
+      .then(function(pingResp){
+        for (var match in pingResp.body) {
           // may be multiple entries.  in that case, pick latest
           if (parseInt(pingResp.body[match].timestamp) > 
             parseInt(timestamp)) {
             timestamp = pingResp.body[match].timestamp;
```

Comment:
actually, timestamp value is used to compare the time value of the previously most recent entry to the current entry.